presto-native-execution/presto_cpp/main/PeriodicMemoryChecker.cpp

```diff
@@ -211,4 +214,22 @@ void PeriodicMemoryChecker::pushbackMemory() {
       kCounterMemoryPushbackLatencyMs, latencyMs * 1000);
   LOG(INFO) << "Shrunk " << velox::succinctBytes(freedBytes);
 }
+
+void startPeriodicMemoryChecker() {
+#ifdef __linux__
+  PeriodicMemoryChecker::Config config;
```

Comment:
Move these lines into the LinuxMemoryChecker constructor and call the base constructor explicitly with the constructed config instead of using the initializer list.

And then without arguments one can follow the folly singleton pattern for Prometheus - add it here and adjust the calls to it. If it is NULL then we are not on Linux and if we are then LinuxMemoryChecker constructor was called.